### PR TITLE
Carousel with api

### PR DESCRIPTION
**Pull request by milofultz**
Implemented the carousel with API calls for the related products, the product details, as well as the ratings and other information.

This PR also added a bit of CSS for the card and carousel.

### CODE REVIEW

File: server.js
Comment (NylonEric) [inline server.js:8]: This `axios.defaults.headers.common` will apply the same authorization to all requests, correct? The Amazon s3 API I am working on will be a different auth/credentials. We should see how to make this apply to a list of routes or define individually.

Amazon REST API docs: https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html

File: client/style.css
Comment (NylonEric) [inline client/style.css:478]: * Underscore: I'm not familiar with this style as much as hyphen-separated class names. Is this the BEM (https://en.bem.info/methodology/naming-convention/ )style? It seems clear enough, we should decide on a consistent approach for all css rules.

* spacing after class selector: I mentioned the similar concern with Griselda's CSS PR as most css I've seen doesn't have a space between the class selector and the braces.

File: client/style.css
Comment (NylonEric) [inline client/style.css:471]: Are these extra spaces necessary?

File: client/src/components/Carousel.jsx
Comment (NylonEric) [inline client/src/components/Carousel.jsx:16]: I'm not sure what this means? if the card is an outfit, it will error out on a GET request for related products?

File: client/src/components/Carousel.jsx
Comment (NylonEric) [inline client/src/components/Carousel.jsx:10]: What is this "magic number' of 22122? Is that something we could put in a const above to clarify what it is?

**Review by NylonEric — CHANGES_REQUESTED** [review]
Overall very clean and understandable code. I left several comments about css style which are really about the whole team deciding on a style for spacing and class naming conventions.

The only substantive change I would request is to look at the API key environmental variable middleware. When we combine other widgets we will need at least another route for posting image files to s3 so we'll need to specify that route with a different 'Authorization' property.

File: client/style.css
Comment (milofultz) [reply client/style.css:478]: I use BEM, but am open for whatever.

I've always kept spaces between the selector and the brace for ease of reading/delineation. Space = new thing. e.g. `div>.thing~section#other-thing{` vs `div > .thing ~ section#other-thing {`

File: server.js
Comment (milofultz) [reply server.js:8]: Good call. I'll make a ticket for it. An easy solutions is just to create header objects we can just plop into the `options` space in the axios request.

```js
const atelierHeader = { 
  header: {
    'Authentication': process.env.GITHUB_TOKEN
  }
};

const s3Header = { 
  header: {
    'Authentication': process.env.S3_TOKEN
  }
};
```

Comment (NylonEric) [reply server.js]: That sounds good. re-request and I'll approve.

File: client/src/components/Carousel.jsx
Comment (NylonEric) [reply client/src/components/Carousel.jsx:10]: Sounds good.

File: client/src/components/Carousel.jsx
Comment (NylonEric) [reply client/src/components/Carousel.jsx:16]: Ok, that makes sense.

File: client/style.css
Comment (NylonEric) [reply client/style.css:476]: Great. We discussed this and decided to adopt a space after class selectors so this looks good.

File: client/src/components/Card.jsx
Comment (NylonEric) [inline client/src/components/Card.jsx:4]: This looks good, you may have a merge conflict with the order of import statements so just fix locally and merge as long as it works on your end.

File: server.js
Comment (NylonEric) [inline server.js:9]: This solution looks like it will work for handling our different authorization needs.

File: server.js
Comment (NylonEric) [inline server.js:2]: This looks like a good solution for storing credentials. Let's make sure all of the team has set this up successfully and is getting the expected results in the test environment.

It might be good to explicitly add a test for this and the API calls that depend on it as well.

**Review by NylonEric — APPROVED** [review]
We've spoken about the previous comments in the first iteration so this all looks good. Be sure to double check that everything still works after you resolve the merge conflict since there have been some changes to the server.js imports since you started this PR.

**Pull request merged**
